experiments/distributedshell/src/main/scala/org/apache/gearpump/distributedshell/DistShellAppMaster.scala

```diff
@@ -17,25 +17,21 @@
  */
 package org.apache.gearpump.distributedshell
 
-import akka.actor.{Props, ActorSystem}
-import akka.testkit.TestProbe
-import org.apache.gearpump.cluster.{TestUtil, UserConfig}
-import org.apache.gearpump.experiments.cluster.task.TaskContext
-import org.scalatest.{WordSpec, Matchers}
-import sys.process._
+import akka.actor.{Actor, ActorRef}
+import org.apache.gearpump.cluster.{AppMasterContext, Application}
+import org.apache.gearpump.experiments.cluster.appmaster.AbstractAppMaster
+import org.slf4j.{LoggerFactory, Logger}
 
-class ShellTaskSpec extends WordSpec with Matchers {
+class DistShellAppMaster(appContext : AppMasterContext, app : Application) extends AbstractAppMaster(appContext, app) {
+  private val LOG: Logger = LoggerFactory.getLogger(getClass)
+  override protected def executorClass: Class[_ <: Actor] = classOf[ShellExecutor]
 
-  "The ShellTask" should {
-    "execute the shell command and return the result" in {
-      val system = ActorSystem("ShellTask", TestUtil.DEFAULT_CONFIG)
-      val mockMaster = TestProbe()(system)
-      val taskContext = TaskContext(0, 0, mockMaster.ref)
-      val shellTask = system.actorOf(Props(classOf[ShellTask], taskContext, UserConfig.empty))
-      shellTask.tell(ShellCommand("ls", "/"), mockMaster.ref)
+  override def onStart(): Unit = {
+    LOG.info(s"Distributed Shell AppMaster started")
+    launchExecutorOnEachWorker()
+  }
 
-      mockMaster.expectMsgType[String]
-      system.shutdown()
-    }
+  override def onExecutorStarted(executor: ActorRef, executorId: Int, workerId: Int): Unit = {
```

Comment:
This is not good. 

An Appmaster at least have a receive handler to receive messages, hiding that handler from user does no good.